pkg/es/writer/esBulkHandler.go

```diff
@@ -330,21 +386,37 @@ func ProcessIndexRequest(rawJson []byte, tsNow uint64, indexNameIn string,
 	bytesReceived uint64, flush bool, localIndexMap map[string]string, myid uint64,
 	rid uint64, idxToStreamIdCache map[string]string,
 	cnameCacheByteHashToStr map[uint64]string, jsParsingStackbuf []byte) error {
+	return nil
+}
+
+func ProcessIndexRequestPle(tsNow uint64, indexNameIn string,
+	bytesReceived uint64, flush bool, localIndexMap map[string]string, myid uint64,
+	rid uint64, idxToStreamIdCache map[string]string,
+	cnameCacheByteHashToStr map[uint64]string, jsParsingStackbuf []byte, pleArray []*writer.ParsedLogEvent) error {
+
+	for _, ple := range pleArray {
+		if ple.GetIndexName() != indexNameIn {
+			return utils.TeeErrorf("ProcessIndexRequestPle: indexName mismatch; want %v, got %v",
+				indexNameIn, ple.GetIndexName())
+		}
+	}
 
 	indexNameConverted := AddAndGetRealIndexName(indexNameIn, localIndexMap, myid)
-	cfgkey := config.GetTimeStampKey()
+	tsKey := config.GetTimeStampKey()
 
 	var docType segment.SIGNAL_TYPE
 	if strings.HasPrefix(indexNameConverted, "jaeger-") {
 		docType = segment.SIGNAL_JAEGER_TRACES
-		cfgkey = "startTimeMillis"
+		tsKey = "startTimeMillis"
 	} else {
 		docType = segment.SIGNAL_EVENTS
 	}
 
-	ts_millis := utils.ExtractTimeStamp(rawJson, &cfgkey)
-	if ts_millis == 0 {
-		ts_millis = tsNow
+	for _, ple := range pleArray {
```

Comment:
nit: do we need to do array walking here ? because later in the flow, we have to walk the array anyways, so why not do it just once .
(this comment does not need to be handled in this PR, just adding here for later reference)

Reply:
Ok, let's do that in a separate PR